partner_email_check/models/res_partner.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+import re
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.constrains('email')
+    def constrains_email(self):
+        for rec in self:
+            self.email_check(rec.email)
+
+    @api.model
+    def email_check(self, email):
+        if re.match(r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)",
```

Comment:
Emails are more complex than you think; [better use `validate_email` like `auth_signup_validate_email` does.](https://github.com/OCA/server-tools/tree/10.0/auth_signup_verify_email#installation)